crates/hyperqueue/src/common/parser2.rs

```diff
@@ -230,12 +231,24 @@ pub fn parse_u32() -> impl CharParser<u32> {
     })
 }
 
-/// Parse 8-byte integer.
-pub fn parse_u64() -> impl CharParser<u64> {
-    parse_integer_string().try_map(|p, span| {
-        p.parse::<u64>()
-            .map_err(|_| ParseError::custom(span, "Cannot parse as 8-byte unsigned integer"))
-    })
+pub fn parse_resource_amount() -> impl CharParser<ResourceAmount> {
```

Comment:
Could you please add a comment above this function that explains what does this parse, along with an example of how could a valid input look like?